### PR TITLE
Fix ScriptEngine not properly resolving paths

### PR DESCRIPTION
**Pull request by huffman**
This would need more testing, but it should fix issues related to a few open bugs. I'll comment on those bugs.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3fc8722f07d34605a870c94bc3a47be2aeb39957 available at: https://gist.github.com/b3c4475186a1fc0ecf35f92f39be87f8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-3fc8722.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-3fc8722.dmg

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <QtNetwork/QNetworkRequest>
 #include <QtNetwork/QNetworkReply>
 
+#include <QtScript/QScriptContextInfo>
```
Comment (howard-stearns) [inline libraries/script-engine/src/ScriptEngine.cpp]: Nice! As you say, needs testing, but this idea looks like a big win!

**Comment by howard-stearns** [comment]
See also https://highfidelity.fogbugz.com/f/cases/1583/scripts-resource-loading-wrong-after-change-from-Script-load-to-Script-include

**Comment by huffman** [comment]
To test:

**Run script to confirm that paths are resolved correctly**
URL: https://hifi-public.s3.amazonaws.com/huffman/resolveTest/a.js

You should see output like this in your log:

```
[10/04 10:01:02] [DEBUG] [hifi.scriptengine] script:print()<< This should be relative to a.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/someFile.js
[10/04 10:01:02] [DEBUG] [hifi.scriptengine] script:print()<< This should be relative to b/b.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/b/someFileB.js
```

On old builds you will see that someFileB.js is not resolved correctly:

```
[10/04 09:57:46] [DEBUG] script:print()<< This should be relative to a.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/someFile.js
[10/04 09:57:46] [DEBUG] script:print()<< This should be relative to b/b.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/someFileB.js
```

**Run entity script to confirm that paths are resolved correctly**
Add to entity: https://hifi-public.s3.amazonaws.com/huffman/resolveTest/aEntity.js

You should see output like this in your log:

```
[10/04 10:03:02] [DEBUG] [hifi.scriptengine] script:print()<< This should be relative to a.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/someFile.js
[10/04 10:03:02] [DEBUG] [hifi.scriptengine] script:print()<< This should be relative to b/b.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/b/someFileB.js
```

**Make sure the URLs printed are correct!**

Lastly, do a general smoke test of defaultScripts, other scripts you might use, and entities with scripts from the marketplace.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3be3331449c169b4bb0c8e614795f840da9b61b available at: https://gist.github.com/b3c4475186a1fc0ecf35f92f39be87f8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-e3be333.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-e3be333.dmg

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 2]
```diff
@@ -1323,7 +1329,7 @@ void ScriptEngine::entityScriptContentAvailable(const EntityItemID& entityID, co
 
     auto scriptCache = DependencyManager::get<ScriptCache>();
     bool isFileUrl = isURL && scriptOrURL.startsWith("file://");
-    auto fileName = QString("(EntityID:%1, %2)").arg(entityID.toString(), isURL ? scriptOrURL : "EmbededEntityScript");
```
Comment (Atlante45) [inline libraries/script-engine/src/ScriptEngine.cpp]: Why remove the entity ID? This can be very useful to figure out troublesome entities.

Comment (huffman) [inline libraries/script-engine/src/ScriptEngine.cpp]: It clobbers the filename in the stacktrace, which this change relies on. I'd be open to putting it somewhere else if there's a good spot.

Comment (Atlante45) [inline libraries/script-engine/src/ScriptEngine.cpp]: I guess we could directly add it to the `hasCorrectSyntax` and `hadUncaughtExceptions` functions.
That would clean things up a bit.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e3f4b705f26e45593c313995882033a2e26c6e0 available at: https://gist.github.com/b3c4475186a1fc0ecf35f92f39be87f8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-8e3f4b7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-8e3f4b7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: df092544371e4bb9e399e01d46fa44ca9766ca89 available at: https://gist.github.com/b3c4475186a1fc0ecf35f92f39be87f8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-df09254.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-df09254.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by hifi-gustavo** [comment]
Build console output for commit: 42b0d8230c60219e5757754df5c9a20dd0ea72b6 available at: https://gist.github.com/b3c4475186a1fc0ecf35f92f39be87f8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-42b0d82.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8664-42b0d82.dmg

**Comment by thoys** [comment]
## QA Testing report


- The Audio script on the Home Server still works 👍  (not sure if it had any includes, but I double checked)

### Interface script
```
[10/06 19:05:36] [DEBUG] Done downloading script at: "https://hifi-public.s3.amazonaws.com/huffman/resolveTest/a.js"
[10/06 19:05:36] [DEBUG] Loading script at  QUrl("https://hifi-public.s3.amazonaws.com/huffman/resolveTest/b/b.js")
[10/06 19:05:36] [DEBUG] script:print()<< This should be relative to a.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/someFile.js
[10/06 19:05:36] [DEBUG] script:print()<< This should be relative to b/b.js :  https://hifi-public.s3.amazonaws.com/huffman/resolveTest/b/someFileB.js
```
- The somefile B script is in the correct folder 👍 

### Entity scripts
```
-

**Pull request merged**